pkg/flexvolume/flexvolume.go

```diff
@@ -141,129 +141,232 @@ func processOpts(optsStr string) (Options, error) {
+		if !ok {
+			return Failf("key '%s' not found in attach options:%#v",
+				OptionKeypvOrVolumeName,
+				opts)
 		}
+
 		nodeName := args[3]
-		ExitWithResult(driver.IsAttached(opts, nodeName))
+		driver := ClaimVolume(volumeID, drivers)
+		if driver == nil {
+			return Failf("Unable to find flexdriver for volume id '%s'", volumeID)
+		}
+
+		return driver.IsAttached(opts, nodeName)
 
 	// <driver executable> mountdevice <mount dir> <mount device> <json options>
 	case "mountdevice":
 		if len(args) != 5 {
-			ExitWithResult(Fail("mountdevice expected exactly 5 arguments; got ", args))
+			return Failf("mountdevice expected exactly 5 arguments; got ", args)
 		}
 
 		mountDir := args[2]
 		mountDevice := args[3]
 
 		opts, err := processOpts(args[4])
 		if err != nil {
-			ExitWithResult(Fail(err))
+			return Failf(err.Error())
 		}
 
-		ExitWithResult(driver.MountDevice(mountDir, mountDevice, opts))
+		volumeID, ok := opts[OptionKeypvOrVolumeName]
+		if !ok {
+			return Failf("key '%s' not found in attach options",
+				OptionKeypvOrVolumeName)
+		}
+
+		driver := ClaimVolume(volumeID, drivers)
+		if driver == nil {
+			return Failf("Unable to find flexdriver for volume id '%s'", volumeID)
+		}
+
+		return driver.MountDevice(mountDir, mountDevice, opts)
 
 	// <driver executable> unmountdevice <mount dir>
 	case "unmountdevice":
 		if len(args) != 3 {
-			ExitWithResult(Fail("unmountdevice expected exactly 3 arguments; got ", args))
+			return Failf("unmountdevice expected exactly 3 arguments; got ", args)
 		}
 
 		mountDir := args[2]
-		ExitWithResult(driver.UnmountDevice(mountDir))
+
+		// assumes that the last part of mountDir is the volumeID
+		parts := strings.Split(mountDir, "/")
+		volumeID := parts[len(parts)-1]
+
+		driver := ClaimVolume(volumeID, drivers)
+		if driver == nil {
+			return Failf("Unable to find flexdriver for volume id '%s'", volumeID)
+		}
+
+		return driver.UnmountDevice(mountDir)
 
 	// <driver executable> mount <mount dir> <json options>
 	case "mount":
 		if len(args) != 4 {
-			ExitWithResult(Fail("mount expected exactly 4 arguments; got ", args))
+			return Failf("mount expected exactly 4 arguments; got ", args)
 		}
 
 		mountDir := args[2]
 
 		opts, err := processOpts(args[3])
 		if err != nil {
-			ExitWithResult(Fail(err))
+			return Failf(err.Error())
 		}
 
-		ExitWithResult(driver.Mount(mountDir, opts))
+		volumeID, ok := opts[OptionKeypvOrVolumeName]
+		if !ok {
+			return Failf("key '%s' not found in attach options",
+				OptionKeypvOrVolumeName)
+		}
+
+		driver := ClaimVolume(volumeID, drivers)
+		if driver == nil {
+			return Failf("Unable to find flexdriver for volume id '%s'", volumeID)
+		}
+
+		return driver.Mount(mountDir, opts)
 
 	// <driver executable> unmount <mount dir>
 	case "unmount":
 		if len(args) != 3 {
-			ExitWithResult(Fail("mount expected exactly 3 arguments; got ", args))
+			return Failf("mount expected exactly 3 arguments; got ", args)
```

Comment:
These Failf messages are generally inconsistent in terms of capitalisation across the board. Suggest lowercase for all since we're pulling in errors which will be lowercase.  

return Failf("Unable") vs return Failf("mount")